src/pint/pintk/pulsar.py

```diff
@@ -600,12 +600,14 @@ def random_models(self, selected):
             spanMJD = maxallMJD - minallMJD
```

Comment:
Worth adding a few percent of extrapolation?

Reply:
Wait, I see. Shouldn't `extra` be applied here rather than later? So it affects the number-of-TOAs calculation?

Reply:
I decided not to, so that `spanMJD` is the actual span.

Reply:
Well, it's the span of the data, not the span the `linspace` is supposed to cover. So if you're in the one-per-day regime, it allocates one per day and then spreads them farther than that by an amount `extra`. I suppose the intention is that `extra` should usually be small so the difference is subtle, but it seems like one should allocate points based on the span they are actually supposed to cover - that is, more points if you have more `extra`?